### PR TITLE
OpenCL can now also use transformed parameters

### PR DESCRIPTION
**Pull request by SteveBronder**
Allows transformed data that is passed to an opencl capable function to be cached on the GPU

Licensed under BSD3

Do we need additional tests for this? Could add the brms model from #423

### CODE REVIEW

File: src/stan_math_backend/Transform_Mir.ml [hunk 1]
```diff
@@ -564,7 +564,13 @@ let trans_prog (p : Program.Typed.t) =
...
   let translate_to_open_cl stmts =
     if !use_opencl then
-      let data_var_idents = List.map ~f:fst p.input_vars in
+      let decl Stmt.Fixed.({pattern; _}) =
+      match pattern with
+        | Decl d ->
+          Some (d.decl_id)
+        | _ -> None
+      in
+      let data_var_idents = List.filter_map ~f:decl p.prepare_data in
```
Comment (seantalts) [inline src/stan_math_backend/Transform_Mir.ml]: Ah, I think you'll still need the input vars, no?

Comment (seantalts) [reply src/stan_math_backend/Transform_Mir.ml]: (or wait, are those included in prepare data?)

Comment (SteveBronder) [reply src/stan_math_backend/Transform_Mir.ml]: Oh my b I assumed prepare_data had all the data, lemme check this out

Comment (SteveBronder) [reply src/stan_math_backend/Transform_Mir.ml]: Prepare data has both! I'll add that to the opencl test

**Comment by SteveBronder** [comment]
Apologies this looks huge now I updated the OpenCL tests for GLM to also use transformed data. Think everything is hunky dory tho!

**Comment by SteveBronder** [comment]
Cool beans ty!

**Pull request merged**